modules/tests/shared/src/test/scala/io/circe/DecoderSuite.scala

```diff
@@ -763,4 +764,6 @@ class DecoderSuite extends CirceMunitSuite with LargeNumberDecoderTestsMunit {
     assert(result.isInvalid)
     assertEquals(result.swap.toOption.map(_.size), Some(2))
   }
+
+  checkAll("Defer[Decoder]", DeferTests[Decoder].defer[MiniInt])
```

Comment:
Can you add some property tests for recursive datastructures like what was used in the example?  Just to verify further the stack safety and serve as an example for users on how to user defer?

Reply:
I added non-property test, as generating recursive data structures are kind of annoying to write `Gen`s for.

Not impossible though, so if you feel strongly that they need to be property tests, I can take a pass at converting them.

Reply:
I think we need this test to have some depth to it.  We need to have the test ensure the stack safety of the implementation of defer.  Its probably fine to use `List` as the type the way you did, but using a generated list of some significant length should do the trick.

Reply:
Clarification question: are you concerned about the stack safety of the `Defer` instance, or the `Decoder` it's used to write?

The former is tested by the `Defer` laws: 
```
  + Defer[Decoder]: defer.defer Identity 0.156s
  + Defer[Decoder]: defer.defer does not evaluate 0.001s
  + Defer[Decoder]: defer.defer is stack safe 0.319s
  + Defer[Decoder]: defer.defer matches fix 0.429s
```

The later I think may be out of our control.